File: web-service/src/main/java/User/UserRole.java
package User;

import core.Permission;

/**
 *
 * @author marcw
 */
public class UserRole {

	String name,
			description;
	boolean isStaff;

	Permission USER,
			ASSIGNMENT,
			ASSIGNMENT_SUBMIT,
			COURSE,
			COURSE_OFFERING,
			DEPARTMENT,
			GROUP;

}

File: web-service/src/models/user/user-role.ts

/**
 * 
 * @author marcw
 */
export class UserRole {
	name: string;

	description: string;

	isStaff: boolean;

	USER: Permission;

	ASSIGNMENT: Permission;

	ASSIGNMENT_SUBMIT: Permission;

	COURSE: Permission;

	COURSE_OFFERING: Permission;

	DEPARTMENT: Permission;

	POST: Permission;

	COMMENT: Permission;

	POLL: Permission;

	POLL_ANSWER: Permission;

	GROUP: Permission;

	constructor() {
		this.isStaff = false;
	}
}